sdks/java/harness/src/test/java/org/apache/beam/fn/harness/CachesTest.java

```diff
@@ -109,15 +121,27 @@ public void testNestedSubCaches() {
     childOfChild.remove("keyB");
     assertEquals("childB", child.peek("keyB"));
     assertNull(childOfChild.peek("keyB"));
+  }
+
+  @Test
+  public void testClearableCache() {
+    ClearableCache<String, String> cache = new ClearableCache<>(Caches.eternal());
+    testCache(cache);
+  }
 
-    // Test that clearing the middle cache impacts children but not parent
-    parent.put("keyA", "parentA");
-    parent.put("keyB", "parentB");
-    child.clear();
-    assertThat(child.keys(), is(emptyIterable()));
-    assertThat(childOfChild.keys(), is(emptyIterable()));
-    assertEquals("parentA", parent.peek("keyA"));
-    assertEquals("parentB", parent.peek("keyB"));
```

Comment:
Is it not possible to have a clearable subcache anymore?

Reply:
The existing implementation to scan all the keys was too slow so I didn't want people to rely on it. This way the have to be cognizant about having a cache where they take the memory overhead of tracking the keys explicitly,